doc/reference/random/index.rst

```diff
@@ -0,0 +1,96 @@
+.. _random_reference:
+
+Random
+######
+
+.. toctree::
+   :maxdepth: 2
+
+The random API subsystem provides random number generation APIs in both
+cryptographically and non-cryptographically secure instances. Which
+random API to use is based on the cryptographic requirements of the
+random number. The non-cryptographic APIs will return random values
+much faster if non-cryptographic values are needed.
+
+The cryptographically secure random functions shall be compliant to the
+FIPS 140-2 [NIST02]_ recommended algorithms. Hardware based random-number
+generators (RNG) can be used on platforms with appropriate hardware support.
+Platforms without hardware RNG support shall use the CTR-DRBG algorithm.
+The algorithm can be provided by TinyCrypt or mbedTLS depending on
+your application performance and resource requirements.
+
+  .. note::
+
+    The CTR-DRBG generator needs a cryptographically secure entropy
+    source to maintain cryptographic security of the PRNG.
```

Comment:
Instead of "maintain..." how about "The CTR-DRBG generator needs an entropy source to establish and maintain the cryptographic security of the PRNG."

I'm not sure the phrase "cryptographically secure entropy source" is all that meaningful.